lib/dash/mpd_utils.js

```diff
@@ -290,9 +290,10 @@ shaka.dash.MpdUtils.createTimeline = function(
  *
  * @param {?number} periodDuration
  * @param {!Array.<!shaka.media.SegmentReference>} references
+ * @param {!shaka.media.PresentationTimeline} presentationTimeline
  */
 shaka.dash.MpdUtils.fitSegmentReferences = function(
-    periodDuration, references) {
+    periodDuration, references, presentationTimeline) {
```

Comment:
I think you need to pass in both |periodDuration| and |segmentAvailabilityDuration| instead of |presentationTimeline| because |presentationTimeline| is not fully initialized here when the first manifest is parsed (see `parsePeriods_()` callsite in `parseManifest_()`)... alternatively, |presentationTimeline| could be initialized beforehand by refactoring `parsePeriods_()` a bit, but I think the former solution is okay.